internal/wrapnix/wrapper.go

```diff
@@ -72,12 +72,51 @@ func CreateWrappers(ctx context.Context, devbox devboxer) error {
 	return createSymlinksForSupportDirs(devbox.ProjectDir())
 }
 
+// CreateDevboxSymlink creates a symlink to the devbox binary
+//
+// Needed because:
+//
+//  1. The bin-wrappers cannot invoke devbox via the Launcher. The Launcher script
+//     invokes some coreutils commands that may themselves be installed by devbox
+//     and so be bin-wrappers. This causes an infinite loop.
+//
+//     So, the bin-wrappers need to directly invoke the devbox binary.
+//
+//  2. The devbox binary's path will change when devbox is updated. This means
+//     that using absolute paths to the devbox binaries in the bin-wrappers
+//     will result in bin-wrappers invoking older devbox binaries.
+//
+//     So, the bin-wrappers need to use a symlink to the latest devbox binary. This
+//     symlink is updated when devbox is updated.
+func CreateDevboxSymlink() (string, error) {
+	curDir := xdg.CacheSubpath(filepath.Join("devbox", "current"))
+	if err := fileutil.EnsureDirExists(curDir, 0755, false /*chmod*/); err != nil {
+		return "", err
+	}
+	currentDevboxSymlinkPath := filepath.Join(curDir, "devbox")
+
+	devboxBinaryPath, err := os.Executable()
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+
+	// We will always re-create this symlink to ensure correctness.
+	if err := os.Remove(currentDevboxSymlinkPath); err != nil && !errors.Is(err, os.ErrNotExist) {
+		return "", errors.WithStack(err)
+	}
+
+	if err := os.Symlink(devboxBinaryPath, currentDevboxSymlinkPath); err != nil {
```

Comment:
Don't return error if error is `os.ErrExist` (to prevent against race conditions)

Reply:
good call.